implementation/src/main/java/io/smallrye/mutiny/groups/UniCreate.java

```diff
@@ -202,6 +203,63 @@ public <T> Uni<T> future(Supplier<Future<? extends T>> supplier) {
         return new UniCreateFromFuture<>(actual);
     }
 
+    /**
+     * Creates a {@link Uni} from the given {@link Future}.
+     * <p>
+     * The produced {@code Uni} emits the item produced by the {@link Future}.
+     * Because {@link Future#get()} is blocking, creating a {@link Uni} from a {@link Future} requires blocking a thread
+     * until the future produces a value, a failure, a timeout, or the subscriber cancels. As a consequence, a thread from the
+     * {@link Infrastructure#getDefaultExecutor()} is used, and waits until the passed future produces an outcome.
+     * <p>
+     * Cancelling the subscription on the produced {@link Uni} cancels the passed {@link Future}
+     * (calling {@link Future#cancel(boolean)}).
+     * <p>
+     * If the produced future has already been completed (or failed), the produced {@link Uni} sends the item or failure
+     * immediately after subscription. If it's not the case the callbacks of the subscriber are called on the thread used to
+     * wait the result (a thread from the Mutiny infrastructure default executor).
+     * <p>
+     *
+     * @param future the future, must not be {@code null}
+     * @param timeout the future timeout
+     * @param <T> the type of item
+     * @return the produced {@link Uni}
+     */
+    @CheckReturnValue
+    public <T> Uni<T> future(Future<? extends T> future, Duration timeout) {
+        Future<? extends T> actual = ParameterValidation.nonNull(future, "future");
+        return new UniCreateFromFuture<>(() -> actual, timeout);
```

Comment:
There's also a duration validation helper that should be used/tested https://github.com/smallrye/smallrye-mutiny/blob/main/implementation/src/main/java/io/smallrye/mutiny/helpers/ParameterValidation.java#L26

Reply:
Yes, the timeout must not be `null` and be strictly positive.